### ISSUE TITLE
False positive keyword detection in MATLAB M Scripts

### ISSUE THREAD

**Issue by tpajenkamp-dspace** [notepad-plus-plus/notepad-plus-plus#5798]
### Description of the Issue

For M scripts the syntax highlighting and code folding breaks if keywords (case-insensitive) are used as structure field names.

### Steps to Reproduce the Issue

1. Create M file with the following content
``` matlab
function [s] = foo()
   s = struct();
   s.End = 42;
   s.if = 0;
end
```
2. "End" in s.End is highlighted as keyword (note that only "end" is an actual keyword, maybe only in newer MATLAB versions(?))
3, "if" in s.if is highlighted as keyword
4. Code-folding matches "function" with "s.End"
5. Code-folding applies a new bracket for "s.if"

### Expected Behavior

* Field names that happen to be keyword names are not highlighted, especially not if they use different case
* "function" is matched with the last line for code folding and this is the only folding option.

### Actual Behavior

![image](https://user-images.githubusercontent.com/46522266/59608299-594bee80-9115-11e9-8eea-d3f8a363899a.png)

* "End" in s.End is highlighted as keyword (note that only "end" is an actual keyword, maybe only in newer MATLAB versions(?))
* "if" in s.if is highlighted as keyword
* Code-folding matches "function" with "s.End"
* Code-folding starts a new bracket for "s.if"

### Debug Information

Notepad++ v7.7   (64-bit)
Build time : May 19 2019 - 13:05:35
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS : Windows 10 (64-bit)
Plugins : mimeTools.dll NppConverter.dll NppExport.dll NPPJSONViewer.dll NppSaveAsAdmin.dll XMLTools.dll

**Comment by zufuliu** [comment]
bug for 'End' is reported at https://sourceforge.net/p/scintilla/bugs/2112/
It's bad style to use if (or other keywords) as a field/property name.

**Comment by tpajenkamp-dspace** [comment]
Thanks for handling and transferring the issue.
Ideally, any legal code would be handled correctly but I understand that there is not much to gain by treating obscure edge cases I stumbled upon when investigating the real issue. "start"/"end" make a nice pairing, though...

**Comment by zufuliu** [comment]
Case-sensitivity for Matlab keywords was fixed.

**Comment by ArkadiuszMichalski** [comment]
`End` was fixed, but `if` still no, in Notepad++ 7.9 RC we see:

![a](https://user-images.githubusercontent.com/2730894/93711489-fb475b80-fb4e-11ea-81c9-c341184d648e.gif)
